sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -686,6 +709,16 @@ private void SetNameFieldsIfNull()
                         Pipeline.ResponseClassifier,
                         Constants.MaxReliabilityRetries);
 
+                    // if using clientside encryption, wrap the auto-retry stream in a decryptor
+                    // we already return a nonseekable stream; returning a crypto stream is fine
+                    if (UsingClientSideEncryption)
+                    {
+                        stream = await new BlobClientSideDecryptor(ClientSideEncryption)
+                            .ClientSideDecryptInternal(stream, response.Value.Metadata, requestedRange, response.Value.ContentRange, async, cancellationToken).ConfigureAwait(false);
```

Comment:
Two things to consider:
- It seems that `BlobClientSideDecryptor` is stateless in this context. Could be a member (field) in this class instead of storing encryption options.
- It's not far away from here to create `IReadStreamTransformer` and have `BlobClientSideDecryptor : IReadStreamTransformer`

not blocking here, but would be nice to have.